pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -80,9 +80,14 @@ func SetupNetworkInterfaces(vm *v1.VirtualMachine, domain *api.Domain) error {
 }
 
 // a factory to get suitable network interface
-func getNetworkClass(network *v1.Network) (NetworkInterface, error) {
-	if network.Pod != nil {
+func getNetworkClass(iface *v1.Interface, network *v1.Network) (NetworkInterface, error) {
```

Comment:
I envisioned it differently. I think here we just need to get the correct interface, according to the network source type and plug() it. How do we plug it is determined by the interface attributes. (`bridge`, `proxy`, `direct` (macvtap) or `passthrough`) As in the API doc [1]
Since `proxy` is only being used with the pod network source, I think, we should add it a new binding mechanism. I have an example for it in an older PR [2]

[1] https://docs.google.com/document/d/10rXr91aqn8MvVcLgHw33WX8BaQwHPZERp25PHxoZGgw
[2] https://github.com/kubevirt/kubevirt/pull/787/files#diff-d3ed8eb8cca020f37cab207c079f2627R257

Reply:
I saw this pull request but in the plug function there are calls that the slirp interface cant implement, like start dhcp, set Interface Routes, etc (all of it is handle by the qemu process) so I cant use the same binding mechanism the bridge interface used.

Reply:
I think you can just override these methods and keep the relevant methods empty.
Would that work?

Reply:
I can try to do it. I need to change the getCachedInterface function? 
Because the slirp interface configuration is in the qemu command line and not as a regular domain interface.

Reply:
yea, it's tricky. I didn't check, but IIRC libvirt had an interface type="user" to setup slirp, isn't it sufficient?

If not, I suppose the best way forward, in this case, is to separate everything after the getBinding to a setup() method and override this method in SlirpPodInterface. So in the Plug method we will have something like 
```
driver, err := getBinding(iface)
if err != nil {
	return err
}
err = driver.setup()
if err != nil {
	return err
}
``` 

not ideal though.

Reply:
Yes they have an interface type user but you can only config the subnet . [1]
 
I use the qemu command line because I need to add the dnssearchname and the portforward parameters.
Cant find anything in the libvirt documentation to add this parameters as part of the interface configuration.

[1] https://libvirt.org/formatdomain.html#elementsNICSSlirp